From 9dad8fd76406d479df5acee5f32e447fc9b18779 Mon Sep 17 00:00:00 2001
Message-ID: <9dad8fd76406d479df5acee5f32e447fc9b18779.1707763691.git.sharlatanus@gmail.com>
In-Reply-To: <e6295f25eabb9fcc8765561825ab7c9b124c1568.1707763691.git.sharlatanus@gmail.com>
References: <e6295f25eabb9fcc8765561825ab7c9b124c1568.1707763691.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Sat, 10 Feb 2024 17:34:47 +0000
Subject: [PATCH v2 21/30] gnu: go-golang-org-x-term: Move to golang-build.

* gnu/packages/golang.scm (go-golang-org-x-term): Move from here ...
* gnu/packages/golang-build.scm: ... to here.

* gnu/packages/disk.scm: Add (gnu packages golang-build) module.
* gnu/packages/image-viewers.scm: As above.

Change-Id: Ibdb637bb129437ef74a0270fd44edc8804bb3055
---
 gnu/packages/disk.scm          |  1 +
 gnu/packages/golang-build.scm  | 25 +++++++++++++++++++++++++
 gnu/packages/golang.scm        | 24 ------------------------
 gnu/packages/image-viewers.scm |  1 +
 4 files changed, 27 insertions(+), 24 deletions(-)

diff --git a/gnu/packages/disk.scm b/gnu/packages/disk.scm
index 165f0a8a74..b0117d77ce 100644
--- a/gnu/packages/disk.scm
+++ b/gnu/packages/disk.scm
@@ -69,6 +69,7 @@ (define-module (gnu packages disk)
   #:use-module (gnu packages gnome)
   #:use-module (gnu packages gnupg)
   #:use-module (gnu packages golang)
+  #:use-module (gnu packages golang-build)
   #:use-module (gnu packages graphics)
   #:use-module (gnu packages gtk)
   #:use-module (gnu packages guile)
diff --git a/gnu/packages/golang-build.scm b/gnu/packages/golang-build.scm
index 7b8653960d..db94ad2c7b 100644
--- a/gnu/packages/golang-build.scm
+++ b/gnu/packages/golang-build.scm
@@ -4,11 +4,13 @@
 ;;; Copyright © 2020 Leo Famulari <leo@famulari.name>
 ;;; Copyright © 2020 Ryan Prior <rprior@protonmail.com>
 ;;; Copyright © 2020 Vagrant Cascadian <vagrant@debian.org>
+;;; Copyright © 2021 Arun Isaac <arunisaac@systemreboot.net>
 ;;; Copyright © 2021 Ludovic Courtès <ludo@gnu.org>
 ;;; Copyright © 2021 hackeryarn <artemchernyak@gmail.com>
 ;;; Copyright © 2023 Artyom V. Poptsov <poptsov.artyom@gmail.com>
 ;;; Copyright © 2023 Felix Lechner <felix.lechner@lease-up.com>
 ;;; Copyright © 2023 Katherine Cox-Buday <cox.katherine.e@gmail.com>
+;;; Copyright © 2023 Nicolas Graves <ngraves@ngraves.fr>
 ;;; Copyright © 2023 Timo Wilken <guix@twilken.net>
 ;;;
 ;;; This file is part of GNU Guix.
@@ -267,6 +269,29 @@ (define-public go-golang-org-x-sys
 support for low-level interaction with the operating system.")
       (license license:bsd-3))))
 
+(define-public go-golang-org-x-term
+  (package
+    (name "go-golang-org-x-term")
+    (version "0.3.0")
+    (source
+     (origin
+       (method git-fetch)
+       (uri (git-reference
+             (url "https://go.googlesource.com/term")
+             (commit (string-append "v" version))))
+       (file-name (git-file-name name version))
+       (sha256
+        (base32 "16s3d62fgdhiqvcib61s5pwxp08hhrmzx8bdv5zk1w1krjizdarl"))))
+    (build-system go-build-system)
+    (arguments '(#:import-path "golang.org/x/term"))
+    (propagated-inputs
+     (list go-golang-org-x-sys))
+    (home-page "https://pkg.go.dev/golang.org/x/term")
+    (synopsis "Go terminal/console support")
+    (description "@code{term} provides support functions for dealing with
+terminals, as commonly found on Unix systems.")
+    (license license:bsd-3)))
+
 (define-public go-golang-org-x-xerrors
   (let ((commit "5ec99f83aff198f5fbd629d6c8d8eb38a04218ca")
         (revision "0"))
diff --git a/gnu/packages/golang.scm b/gnu/packages/golang.scm
index 8bfabb9da9..240c87c3ba 100644
--- a/gnu/packages/golang.scm
+++ b/gnu/packages/golang.scm
@@ -9659,30 +9659,6 @@ (define-public go-github-com-dustin-go-humanize
 back.")
     (license license:expat)))
 
-(define-public go-golang-org-x-term
-  (package
-    (name "go-golang-org-x-term")
-    (version "0.3.0")
-    (source
-     (origin
-       (method git-fetch)
-       (uri (git-reference
-             (url "https://go.googlesource.com/term")
-             (commit (string-append "v" version))))
-       (file-name (git-file-name name version))
-       (sha256
-        (base32
-         "16s3d62fgdhiqvcib61s5pwxp08hhrmzx8bdv5zk1w1krjizdarl"))))
-    (build-system go-build-system)
-    (arguments '(#:import-path "golang.org/x/term"))
-    (propagated-inputs
-     (list go-golang-org-x-sys))
-    (home-page "https://pkg.go.dev/golang.org/x/term")
-    (synopsis "Go terminal/console support")
-    (description "@code{term} provides support functions for dealing with
-terminals, as commonly found on Unix systems.")
-    (license license:bsd-3)))
-
 (define-public go-github-com-klauspost-compress
   (package
     (name "go-github-com-klauspost-compress")
diff --git a/gnu/packages/image-viewers.scm b/gnu/packages/image-viewers.scm
index 206b91c878..f2b87268f0 100644
--- a/gnu/packages/image-viewers.scm
+++ b/gnu/packages/image-viewers.scm
@@ -77,6 +77,7 @@ (define-module (gnu packages image-viewers)
   #:use-module (gnu packages glib)
   #:use-module (gnu packages gnome)
   #:use-module (gnu packages golang)
+  #:use-module (gnu packages golang-build)
   #:use-module (gnu packages gtk)
   #:use-module (gnu packages graphics)
   #:use-module (gnu packages image)
-- 
2.41.0

